core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala

```diff
@@ -116,7 +116,7 @@ object Instance extends SourceInfoDoc {
     // Check to see if the module is already defined internally or externally
     val existingMod = Builder.components.map {
       case c: DefModule if c.id == definition.proto                 => Some(c)
-      case c: DefBlackBox if c.name == definition.proto.name        => Some(c)
+      case c: DefBlackBox[_] if c.name == definition.proto.name     => Some(c)
```

Comment:
what is going on with this change? Can we add a unit test related to this change?

Reply:
so this is a hacky attempt to distinguish black boxes with `Instance`-type ids here: https://github.com/chipsalliance/chisel/pull/3545/files#diff-dc0f0eefedf62d62714db80d54d7ead1df351b622236533afd6ae5eb4503d710R32

this is not currently working, could use some ideas whether or not it's workable. cc @jackkoenig

Reply:
specifically what's not working is that with this change, the [dedup spec](https://github.com/chipsalliance/chisel/pull/3545/files#diff-362b0d2e1143590082fa00d7896da154acc13e6579842f044d4667a160a500b2R113) is failing to not dedup regular modules in different groups